### DISCUSSION TITLE
Double characters/strange values in table

### DISCUSSION THREAD

**Discussion by jakobdo** [jsvine/pdfplumber#626] [Ask for help with specific PDFs]
Hello again, in this pdf: https://www.taggmbh.at/fileadmin/content/TAG-Website-Content-SM/2016_Maintenance_list_PROD_PDF.pdf
When using the extract tables, on page 2, I get the following values from one of the lines:

00:'TT..0116'
01:'SStaytsiotenm w toerskts'
02:'MCSS  AWrneoiteldnsdteoirnf'
03:'0210..0037..22001166'
04:'0068::0000  hh'
05:'299'
06:'0280..0057..22001166'
07:'0165::0000  hh'
08:'1289'
09:'678 h doauyrss'

etc...

In pdf it looks like:
![image](https://user-images.githubusercontent.com/278270/158545749-8aad8cb2-31c2-42a5-9310-5f6cc70fcda3.png)

Can I fix this issue somehow or is this pdf just buggy?

It seems line the line 1 from page 1 and line 1 from page 2 is mixed.

Page 1 line 1:
![image](https://user-images.githubusercontent.com/278270/158546504-b48099be-58cd-40dd-89b5-87f1c91f8cf8.png)

Page 2 line 1:
![image](https://user-images.githubusercontent.com/278270/158546618-8e86ce2d-8b51-41cc-9a0c-ff973f6be643.png)

**Comment by jakobdo** [comment]
Some of the values can be corrected by: row[0][1::2], but this is not 100% perfect for all columns. So I guess this is a "buggy" PDF, but I really hope some of you PDF-experts has a bullet proof solution for this issue. :)

**Reply by jsvine** [reply]
Ah, yes, unfortunately that does seem the PDF is a little buggy — or, perhaps more accurately, poorly designed. Digging in a bit, you can see that there are actually two sets of characters written on top of each other in the same line (though one apparently invisible, possible because a rectangle was written over it):

```python
im = page.to_image()
start = 1541
count = 31
print(pdfplumber.utils.extract_text(page.chars[start:start+count]))
im.reset().draw_rects(page.chars[start:start+count])
```

Printed: `T.01 Station works MS Arnoldstein`
Image:

![image](https://user-images.githubusercontent.com/534702/158666375-0f59d5ca-325e-4645-b4d4-0d94142fd633.png)

... versus:

```python
start = 2645
count = 28
print(pdfplumber.utils.extract_text(page.chars[start:start+count]))
im.reset().draw_rects(page.chars[start:start+count])
```

Printed: `T.01 Station works MS Arnoldstein`

Image:

![image](https://user-images.githubusercontent.com/534702/158666459-3c5c8074-4874-46f0-9af1-a3b14caf5807.png)

---

There *might* be ways of identifying and removing the errant/extra text, but it would probably require some logic fairly specific to this PDF and the objects within it.

**Reply by jakobdo** [reply]
Thanks again @jsvine . Not sure how to add a logic that will fix this. But thanks again for your time and this nice library.
